index.js

```diff
@@ -86,117 +53,111 @@ function queryContainers(details) {
+    if (!(convertedIdentity.userContextId in identitiesState)) {
+      identitiesState[convertedIdentity.userContextId] = {hiddenTabUrls: []};
     }
   });
 
   return Promise.resolve(identities);
 }
 
-function createContainer(details) {
-  const identity = ContextualIdentityService.create(details.name,
-                                                  details.icon,
-                                                  details.color);
-
-  return Promise.resolve(convert(identity));
-}
-
-function updateContainer(cookieStoreId, details) {
-  const containerId = getContainerForCookieStoreId(cookieStoreId);
-
-  if (!containerId) {
+function removeContainer(userContextId) {
+  if (!userContextId) {
     return Promise.resolve(null);
   }
 
-  const identity = ContextualIdentityService.getIdentityFromId(containerId);
+  const identity = ContextualIdentityService.getIdentityFromId(userContextId);
 
   if (!identity) {
     return Promise.resolve(null);
   }
 
-  if (details.name !== null) {
-    identity.name = details.name;
-  }
+  // We have to create the identity object before removing it.
+  const convertedIdentity = convert(identity);
 
-  if (details.color !== null) {
-    identity.color = details.color;
+  if (!ContextualIdentityService.remove(identity.userContextId)) {
+    return Promise.resolve(null);
   }
 
-  if (details.icon !== null) {
-    identity.icon = details.icon;
+  return Promise.resolve(convertedIdentity);
+}
+
+function openTab(args) {
+  let browserWin = Services.wm.getMostRecentWindow('navigator:browser');
+
+  // This should not really happen.
+  if (!browserWin || !browserWin.gBrowser) {
+    return Promise.resolve(false);
   }
 
-  if (!ContextualIdentityService.update(identity.userContextId,
-                                        identity.name, identity.icon,
-                                        identity.color)) {
-    return Promise.resolve(null);
+  let userContextId = 0;
+  if ('userContextId' in args) {
+    userContextId = args.userContextId;
   }
 
-  return Promise.resolve(convert(identity));
+  let tab = browserWin.gBrowser.addTab(args.url || null,
+                                       { userContextId: userContextId })
+  browserWin.gBrowser.selectedTab = tab;
+  return Promise.resolve(true);
 }
 
-function removeContainer(cookieStoreId) {
-  const containerId = getContainerForCookieStoreId(cookieStoreId);
+function queryTabs(args) {
+  return new Promise((resolve, reject) => {
+    let tabList = [];
 
-  if (!containerId) {
-    return Promise.resolve(null);
-  }
+    for (let tab of tabs) {
+      let xulTab = viewFor(tab);
+      let userContextId = parseInt(xulTab.getAttribute('usercontextid') || 0, 10);
 
-  const identity = ContextualIdentityService.getIdentityFromId(containerId);
+      if ("userContextId" in args && args.userContextId != userContextId) {
+        continue;
+      }
 
-  if (!identity) {
-    return Promise.resolve(null);
-  }
+      tabList.push({
+        id: tab.id,
```

Comment:
this `tab.id` had a weird value when I spot-checked this branch. The values was `-3-9` ? So when the sort algorithm hit it, I got `Error: Incorrect argument types for tabs.move.`